lib/std/child_process.zig

```diff
@@ -656,7 +656,94 @@ pub const ChildProcess = struct {
         }
     }
 
+    pub const Forked = struct {
+        child: *ChildProcess,
+        stdin_pipe: [2]os.fd_t,
+        stdout_pipe: [2]os.fd_t,
+        stderr_pipe: [2]os.fd_t,
+        dev_null_fd: os.fd_t,
+        err_pipe_write: os.fd_t,
+        argv_buf: [:null]?[*:0]u8,
+        envp: [*:null]?[*:0]u8,
+
+        pub fn reportError(self: Forked, err: ChildProcess.SpawnError) noreturn {
+            forkChildErrReport(self.err_pipe_write, err);
+        }
+
+        pub fn setup(self: Forked) !void {
+            try setUpChildIo(self.child.stdin_behavior, self.stdin_pipe[0], os.STDIN_FILENO, self.dev_null_fd);
+            try setUpChildIo(self.child.stdout_behavior, self.stdout_pipe[1], os.STDOUT_FILENO, self.dev_null_fd);
+            try setUpChildIo(self.child.stderr_behavior, self.stderr_pipe[1], os.STDERR_FILENO, self.dev_null_fd);
+
+            if (self.child.stdin_behavior == .Pipe) {
+                os.close(self.stdin_pipe[0]);
+                os.close(self.stdin_pipe[1]);
+            }
+            if (self.child.stdout_behavior == .Pipe) {
+                os.close(self.stdout_pipe[0]);
+                os.close(self.stdout_pipe[1]);
+            }
+            if (self.child.stderr_behavior == .Pipe) {
+                os.close(self.stderr_pipe[0]);
+                os.close(self.stderr_pipe[1]);
+            }
+
+            if (self.child.cwd_dir) |cwd| {
+                try os.fchdir(cwd.fd);
+            } else if (self.child.cwd) |cwd| {
+                try os.chdir(cwd);
+            }
+
+            if (self.child.gid) |gid| {
+                try os.setregid(gid, gid);
+            }
+
+            if (self.child.uid) |uid| {
+                try os.setreuid(uid, uid);
+            }
+        }
+
+        pub fn exec(self: Forked) os.ExecveError {
+            return switch (self.child.expand_arg0) {
+                .expand => os.execvpeZ_expandArg0(.expand, self.argv_buf.ptr[0].?, self.argv_buf.ptr, self.envp),
+                .no_expand => os.execvpeZ_expandArg0(.no_expand, self.argv_buf.ptr[0].?, self.argv_buf.ptr, self.envp),
+            };
+        }
+    };
+
+    /// fork off the child process.  Calling fork is equivalent to spawn except it will return
+    /// in the child process before calling setup/exec.  This provides an opportunity for the
+    /// caller to perform additional setup in the child process before calling setup/exec.
+    ///
+    /// This API does not support resource cleanup in the child process after fork returns.
+    /// The caller is only mean to perform setup and then call exec or exit afterwards.
+    ///
+    /// Note there are limitations on what can be done in the child process after fork returns.
+    /// For example, calling malloc from libc may cause deadlock.
```

Comment:
This is missing a description of the return values `null` and `Forked` from `forkPosix`: This returns \`null\` for the child and \`Forked\` for the parent`.

Reply:
Good catch, fixed